core/src/main/java/org/testcontainers/images/ParsedDockerfile.java

```diff
@@ -0,0 +1,67 @@
+package org.testcontainers.images;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Representation of a Dockerfile, with partial parsing for extraction of a minimal set of data.
+ */
+@Slf4j
+public class ParsedDockerfile {
```

Comment:
does it need to be public? Maybe we can start with an internal one first and make it public only when we feel confident about it :)

Reply:
Unfortunately it's used by both `org.testcontainers.images.builder.ImageFromDockerfile` and `org.testcontainers.containers.DockerComposeContainer`, so we can't make it package private.